extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -815,8 +817,15 @@ class CornerstoneViewportService extends PubSubService implements IViewportServi
     const displaySetUIDs = viewportGridService.getDisplaySetsUIDsForViewport(viewport.id);
     const displaySet = displaySetService.getDisplaySetByUID(displaySetUIDs[0]);
     const displaySetModality = displaySet?.Modality;
+
+    // filter the segmentation modalities since the overlay will get handled below via the segmentation service
+    const filteredVolumeInputArray = volumeInputArray.filter(volumeInput => {
+      const displaySet = displaySetService.getDisplaySetByUID(volumeInput.displaySetInstanceUID);
+      return !['SEG', 'RTSTRUCT'].includes(displaySet?.Modality);
```

Comment:
Can you use the parameter for the colorbar here - if you called it something like displaySet.isOverlayImage then it can be handled without embedding the knowledge of modality names.

Reply:
That same key can probably be used for the colorbar presence/absence control.